internal/adminapi/client.go

```diff
@@ -67,6 +71,32 @@ func MakeHTTPClient(opts *HTTPClientOpts) (*http.Client, error) {
 		}
 		tlsConfig.RootCAs = certPool
 	}
+
+	if opts.TLSClientCertPath != "" && opts.TLSClientKeyPath != "" {
+		tlsClientCertPath := opts.TLSClientCertPath
+		tlsClientCert, err := os.ReadFile(tlsClientCertPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read kong-admin-tls-client-cert-file from path '%s': %w", tlsClientCertPath, err)
+		}
+		opts.TLSClientCert = string(tlsClientCert)
+
+		tlsClientKeyPath := opts.TLSClientKeyPath
+		tlsClientKey, err := os.ReadFile(tlsClientKeyPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read kong-admin-tls-client-key-file from path '%s': %w", tlsClientKeyPath, err)
+		}
+		opts.TLSClientKey = string(tlsClientKey)
```

Comment:
Because we now accept both the certificate and key (i.e. `TLSClientCert`, `TLSClientKey`) as well as file paths (i.e. `TLSClientCertPath`, `TLSClientKeyPath`) we've effectively implemented an override that is not documented or obvious to the user. If a user manages to provide both `TLSClientCert` and `TLSClientCertPath` simultaneously we simply ignore the former and take the latter but can we be sure that's the right choice?

Suggestion: fail and provide a clear error to the caller that either one or the other needs to be defined, not both.

Reply:
Will fix accordingly.